students/lisa_ferrier/lesson03/mailroom.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# mailroom.py
+# Lisa Ferrier, Python 210, Lesson 03
+
+
+import sys
+from operator import itemgetter
+
+donations = [("William Gates, III", [653772.32, 12.17]),
+            ("Jeff Bezos", [877.33]),
+            ("Paul Allen", [663.23, 43.87, 1.32]),
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),
+            ("Oprah Winfrey", [15000.00, 12500.00]),
+             ]
+
+
+def user_options():
+    while True:
+        response = input('''
+                     1 - Send a Thank You
+                     2 - Create a Report
+                     3 - Quit.
+                     Select option 1, 2, or 3.
+                     ''')
+        if response == "1":
+            print("Okay, let's send a thank you.")
+            send_thank_you()
+        elif response == "2":
+            print("Okay, here's the report:")
+            create_report()
+        elif response == "3":
+            quit_program()
+        else:
+            print("That's not a valid option. Please enter 1, 2, or 3.")
+
+
+def send_thank_you():
+    while True:
+        donor_name = input("\nEnter the donor's full name. Enter 'list' to see the complete donation list. Enter 'home' to return to main menu.\n")
+        donor_name = donor_name.title()
+        if donor_name == 'Home':
+            user_options()
+        elif donor_name == "List":
+            for d in donations:
+                print(d[0])
+        else:
+            for d in donations:
+                if donor_name == d[0]:
+                    donation_amount = sum(d[1])
+                    break
+            else:
+                donation_amount = float(input("Donor not found. Enter donation amount."))
+                donations.append([donor_name, [donation_amount]])
+            print('''
+Dear {},
+Thank you for your generous donation in the amount of {:.2f}.
+Sincerely,
+This charity.
+                  '''.format(donor_name, donation_amount))
```

Comment:
Unable to add additional donations to existing donors - might take another look at this.